File: bench.js
#!/usr/bin/nodejs

function prime(n){
	if(n == 2) return true;
	if(n < 2) return false;
	if((n & 1) == 0) return 0;
	for(var i = 3; i < n; i += 2)
		if(n % i == 0)
			return false;
	return true;
}
function primes(n, buf){
	if(n <= 0) return;
	buf[0] = 2;
	var i = 1, j = 3;
	while(i < n){
		if(prime(j)){
			buf[i] = j;
			i++;
		}
		j += 2;
	}
}
function millis(){
	return new Date().getTime();
}

var amount = 1000;
var benches = 1;
if(process.argv.length > 2)
	amount = process.argv[2];
if(process.argv.length > 3)
	benches = process.argv[3];

var buf = new Array(amount);
var times = new Array(benches);
for(var i = 0; i < benches - 1; i++){
	var start = millis();
	primes(amount, buf);
	times[i] = millis() - start;
}
var s = "";
if(benches == 1)
	s = times[0] + "ms";
else{
	s = "[" + times[0];
	for(var i = 0; i < times.length; i++)
		s += ", " + times[i];
	s += "]";
}
console.log("[NodeJS] Finding " + amount + " primes took " + s);

File: bench.py
#!/usr/bin/python3
def prime(n):
	if n == 2:
		return True
	if (n & 1) == 0 or n < 2:
		return False
	for i in range(3, n, 2):
		if n % i == 0:
			return False
	return True
def primes(buf):
	primes(buf, len(buf))
def primes(buf, n):
	if n < 1: return
	j = 1
	i = 1
	buf[0] = 2
	while i < n:
		if prime(j):
			buf[i] = j
			i = i + 1
		j = j + 2

if __name__ == "__main__":
	import time, sys
	amount = 1000
	benches = 1
	if len(sys.argv) > 1:
		amount = int(sys.argv[1])
	if len(sys.argv) > 2:
		benches = int(sys.argv[2])
	def millis():
		return int(time.time() * 1000.0)
	buf = [0] * amount
	times = [0] * benches
	for i in range(0, benches):
		start = millis()
		primes(buf, amount)
		times[i] = millis() - start
	if len(times) == 1:
		print("[Python] Finding %i primes took %ims" % (amount, times[i]))
	else:
		print("[Python] Finding %i primes took %s" % (amount, str(times)))
